src/modules/cpu/kernel/phase.hpp

```diff
@@ -0,0 +1,965 @@
+            srcPtr2RowB = srcPtr2RowG + srcDescPtr->strides.cStride;
+            dstPtrRow = dstPtrChannel;
+
+            for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+            {
+                Rpp8u *srcPtr1TempR, *srcPtr1TempG, *srcPtr1TempB, *srcPtr2TempR, *srcPtr2TempG, *srcPtr2TempB, *dstPtrTemp;
+                srcPtr1TempR = srcPtr1RowR;
+                srcPtr1TempG = srcPtr1RowG;
+                srcPtr1TempB = srcPtr1RowB;
+                srcPtr2TempR = srcPtr2RowR;
+                srcPtr2TempG = srcPtr2RowG;
+                srcPtr2TempB = srcPtr2RowB;
+                dstPtrTemp = dstPtrRow;
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += 16)
+                {
+                    __m256 p1[6], p2[6];
+
+                    rpp_simd_load(rpp_load48_u8pln3_to_f32pln3_avx, srcPtr1TempR, srcPtr1TempG, srcPtr1TempB, p1);    // simd loads
+                    rpp_simd_load(rpp_load48_u8pln3_to_f32pln3_avx, srcPtr2TempR, srcPtr2TempG, srcPtr2TempB, p2);    // simd loads
+                    p1[0] = _mm256_round_ps(_mm256_mul_ps(atan2_ps(p1[0], p2[0]), pMul), _MM_FROUND_TO_ZERO);    // phase computation
+                    p1[1] = _mm256_round_ps(_mm256_mul_ps(atan2_ps(p1[1], p2[1]), pMul), _MM_FROUND_TO_ZERO);    // phase computation
+                    p1[2] = _mm256_round_ps(_mm256_mul_ps(atan2_ps(p1[2], p2[2]), pMul), _MM_FROUND_TO_ZERO);    // phase computation
+                    p1[3] = _mm256_round_ps(_mm256_mul_ps(atan2_ps(p1[3], p2[3]), pMul), _MM_FROUND_TO_ZERO);    // phase computation
+                    p1[4] = _mm256_round_ps(_mm256_mul_ps(atan2_ps(p1[4], p2[4]), pMul), _MM_FROUND_TO_ZERO);    // phase computation
+                    p1[5] = _mm256_round_ps(_mm256_mul_ps(atan2_ps(p1[5], p2[5]), pMul), _MM_FROUND_TO_ZERO);    // phase computation
+                    rpp_simd_store(rpp_store48_f32pln3_to_u8pkd3_avx, dstPtrTemp, p1);    // simd stores
+
+                    srcPtr1TempR += 16;
+                    srcPtr1TempG += 16;
+                    srcPtr1TempB += 16;
+                    srcPtr2TempR += 16;
+                    srcPtr2TempG += 16;
+                    srcPtr2TempB += 16;
+                    dstPtrTemp += 48;
+                }
+                for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                {
+                    dstPtrTemp[0] = (Rpp8u) RPPPIXELCHECK(atan(((Rpp32f) (*srcPtr1TempR)) / ((Rpp32f) (*srcPtr2TempR))) * multiplier);
+                    dstPtrTemp[1] = (Rpp8u) RPPPIXELCHECK(atan(((Rpp32f) (*srcPtr1TempG)) / ((Rpp32f) (*srcPtr2TempG))) * multiplier);
+                    dstPtrTemp[2] = (Rpp8u) RPPPIXELCHECK(atan(((Rpp32f) (*srcPtr1TempB)) / ((Rpp32f) (*srcPtr2TempB))) * multiplier);
+
+                    srcPtr1TempR++;
+                    srcPtr1TempG++;
+                    srcPtr1TempB++;
+                    srcPtr2TempR++;
```

Comment:
You can post-increment in L174-176 itself

Reply:
This post increment couldn't be done inline as it was fed as input to atan thus the effect of post increment was defeated and replaced by a normal increment and lead to output mismatches.

Reply:
Thus used post increment wherever possible(for destination pointer)